python/GafferSceneTest/EditScopeAlgoTest.py

```diff
@@ -0,0 +1,183 @@
+		self.assertTrue( GafferScene.EditScopeAlgo.getPruned( scope, "/group/plane" ) )
+		self.assertFalse( GafferScene.EditScopeAlgo.getPruned( scope, "/group/cube" ) )
+		self.assertEqual( len( list( GafferScene.SceneProcessor.Range( scope ) ) ), 1 )
+		self.assertEqual( scope["PruningEdits"]["paths"].getValue(), IECore.StringVectorData( [ "/group/plane" ] ) )
+		self.assertFalse( GafferScene.SceneAlgo.exists( scope["out"], "/group/plane" ) )
+		self.assertTrue( GafferScene.SceneAlgo.exists( scope["out"], "/group/cube" ) )
+
+		GafferScene.EditScopeAlgo.setPruned( scope, IECore.PathMatcher( [ "/group/plane", "/group/cube" ] ), True )
+		self.assertTrue( GafferScene.EditScopeAlgo.getPruned( scope, "/group/plane" ) )
+		self.assertTrue( GafferScene.EditScopeAlgo.getPruned( scope, "/group/cube" ) )
+		self.assertEqual( len( list( GafferScene.SceneProcessor.Range( scope ) ) ), 1 )
+		self.assertEqual( scope["PruningEdits"]["paths"].getValue(), IECore.StringVectorData( [ "/group/cube", "/group/plane" ] ) )
+		self.assertFalse( GafferScene.SceneAlgo.exists( scope["out"], "/group/plane" ) )
+		self.assertFalse( GafferScene.SceneAlgo.exists( scope["out"], "/group/cube" ) )
+
+		GafferScene.EditScopeAlgo.setPruned( scope, IECore.PathMatcher( [ "/group/plane", "/group/cube" ] ), False )
+		self.assertFalse( GafferScene.EditScopeAlgo.getPruned( scope, "/group/plane" ) )
+		self.assertFalse( GafferScene.EditScopeAlgo.getPruned( scope, "/group/cube" ) )
+		self.assertEqual( len( list( GafferScene.SceneProcessor.Range( scope ) ) ), 1 )
+		self.assertEqual( scope["PruningEdits"]["paths"].getValue(), IECore.StringVectorData() )
+		self.assertTrue( GafferScene.SceneAlgo.exists( scope["out"], "/group/plane" ) )
+		self.assertTrue( GafferScene.SceneAlgo.exists( scope["out"], "/group/cube" ) )
+
+	def testPruningSerialisation( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		s["plane"] = GafferScene.Plane()
+
+		s["editScope"] = Gaffer.EditScope()
+		s["editScope"].setup( s["plane"]["out"] )
+
+		GafferScene.EditScopeAlgo.setPruned( s["editScope"], "/plane", True )
+		self.assertTrue( GafferScene.EditScopeAlgo.getPruned( s["editScope"], "/plane" ), True )
+
+		s2 = Gaffer.ScriptNode()
+		s2.execute( s.serialise() )
+
+		self.assertTrue( GafferScene.EditScopeAlgo.getPruned( s2["editScope"], "/plane" ), True )
+
+	def testTransform( self ) :
+
+		plane = GafferScene.Plane()
+		plane["transform"]["translate"].setValue( imath.V3f( 1, 0, 0 ) )
+
+		editScope = Gaffer.EditScope()
+		editScope.setup( plane["out"] )
+		editScope["in"].setInput( plane["out"] )
+
+		self.assertFalse( GafferScene.EditScopeAlgo.hasTransform( editScope, "/plane" ) )
+		self.assertIsNone( GafferScene.EditScopeAlgo.getTransform( editScope, "/plane" ) )
+		self.assertIsNone( GafferScene.EditScopeAlgo.acquireTransformPlugs( editScope, "/plane", createIfNecessary = False ) )
+		self.assertEqual( editScope["out"].transform( "/plane" ), plane["transform"].matrix() )
```

Comment:
Be good to have a `self.assertEqual( len( list( GafferScene.SceneProcessor.Range( scope ) ) ), 0 )` test here too to catch that we're not inadvertently making the Processor.

Reply:
4fa7b71